### ISSUE TITLE
Simulating a circuit with noise based on the Tokyo device causes a MapperError

### ISSUE THREAD

**Issue by bjader** [Qiskit/qiskit-terra#2073]
### Informations

- **Qiskit Aer version**: 0.1.0
- **Python version**: 3.7.2
- **Operating system**: Ubuntu 18.04.2
### What is the current behavior?
When trying to run a circuit with `noise_model` and `coupling_map` based off the Tokyo device, I encounter the following error:
```
INFO:qiskit.transpiler._transpiler:pre-mapping properties: {'size': 54, 'depth': 51, 'width': 5, 'bits': 1, 'factors': 1, 'operations': {'u3': 13, 'cx': 30, 'U': 7, 'u2': 2, 'barrier': 1, 'measure': 1}}
INFO:qiskit.transpiler._transpiler:initial layout: None
Traceback (most recent call last):
...
...
qiskit.mapper._mappererror.MapperError: 'swap_mapper failed: layer 1, sublayer 0, "cx q[4],q[3];\n"'
```
This happens consistently over several attempts.
### Steps to reproduce the problem
Below is an adapted version of the code I am running:
```python
tokyo = IBMQ.get_backend('ibmq_20_tokyo')
aer_simulator = Aer.get_backend('qasm_simulator')

properties = tokyo.properties()
coupling_map = tokyo.configuration().coupling_map
noise_model = noise.device.basic_device_noise_model(properties)
basis_gates = noise_model.basis_gates

qr = QuantumRegister(5)
cr = ClassicalRegister(1)
qc = QuantumCircuit(qr, cr)

hamiltonian = ...
ground_state = hamiltonian.calculate_ground_state()
qc.initialize(ground_state, [qr[1], qr[2], qr[3], qr[4]])

qc.h(qr[0])
qc.x(qr[0])
qc.cx(qr[0], qr[1])
qc.x(qr[0])
qc.cx(qr[0], qr[1])
qc.h(qr[0])

execute(qc, aer_simulator, noise_model=noise_model, coupling_map=coupling_map,basis_gates=basis_gates)
```

The circuit looks as follows:
![image](https://user-images.githubusercontent.com/14994219/54928678-3510b500-4f0c-11e9-8a35-46b35e7b14e2.png)

### What is the expected behavior?
When runnning the same circuit with a noise model based on the Poughkeepsie device, instead of Tokyo, the mapping does not throw an error:
```
NFO:qiskit.transpiler._transpiler:pre-mapping properties: {'size': 54, 'depth': 51, 'width': 5, 'bits': 1, 'factors': 1, 'operations': {'u3': 13, 'cx': 30, 'U': 7, 'u2': 2, 'barrier': 1, 'measure': 1}}
INFO:qiskit.transpiler._transpiler:initial layout: None
INFO:qiskit.transpiler._transpiler:final layout: {(QuantumRegister(5, 'q0'), 0): (QuantumRegister(20, 'q'), 0), (QuantumRegister(5, 'q0'), 1): (QuantumRegister(20, 'q'), 1), (QuantumRegister(5, 'q0'), 2): (QuantumRegister(20, 'q'), 2), (QuantumRegister(5, 'q0'), 3): (QuantumRegister(20, 'q'), 3), (QuantumRegister(5, 'q0'), 4): (QuantumRegister(20, 'q'), 4)}
INFO:qiskit.transpiler._transpiler:post-mapping properties: {'size': 92, 'depth': 82, 'width': 20, 'bits': 1, 'factors': 16, 'operations': {'u3': 13, 'cx': 68, 'U': 7, 'barrier': 2, 'u2': 1, 'measure': 1}}
```
Equally when running the same circuit on the real Tokyo device the mapping does not throw an error:
```
INFO:qiskit.transpiler._transpiler:pre-mapping properties: {'size': 54, 'depth': 51, 'width': 5, 'bits': 1, 'factors': 1, 'operations': {'u3': 13, 'cx': 30, 'u1': 7, 'u2': 2, 'barrier': 1, 'measure': 1}}
INFO:qiskit.transpiler._transpiler:initial layout: {('q0', 0): (QuantumRegister(20, 'q'), 5), ('q0', 1): (QuantumRegister(20, 'q'), 0), ('q0', 2): (QuantumRegister(20, 'q'), 6), ('q0', 3): (QuantumRegister(20, 'q'), 10), ('q0', 4): (QuantumRegister(20, 'q'), 11)}
INFO:qiskit.transpiler._transpiler:final layout: {(QuantumRegister(5, 'q0'), 0): (QuantumRegister(20, 'q'), 5), (QuantumRegister(5, 'q0'), 1): (QuantumRegister(20, 'q'), 0), (QuantumRegister(5, 'q0'), 2): (QuantumRegister(20, 'q'), 6), (QuantumRegister(5, 'q0'), 3): (QuantumRegister(20, 'q'), 10), (QuantumRegister(5, 'q0'), 4): (QuantumRegister(20, 'q'), 11)}
INFO:qiskit.transpiler._transpiler:post-mapping properties: {'size': 72, 'depth': 65, 'width': 20, 'bits': 1, 'factors': 16, 'operations': {'u3': 13, 'cx': 48, 'u1': 7, 'barrier': 2, 'u2': 1, 'measure': 1}}
```
### Suggested solutions
Unsure - am rather stumped that the real Tokyo device does not throw the same error. Is the transpiling different compared to an Aer simulator with identical properties and coupling map?

I should also add that the simple example used in the "Device noise simulation" chapter of `qiskit-tutorial` does not throw the same error when using Tokyo's properties.

**Comment by bjader** [comment]
```python
execute(qc, aer_simulator, coupling_map=coupling_map, noise_model=noise_model)
execute(qc, aer_simulator, coupling_map=coupling_map, basis_gates=basis_gates)
execute(qc, aer_simulator, coupling_map=coupling_map)
```
I can confirm that all the above permutations raise the same error as the original post.
Only if I remove the coupling map completely: 
```python
execute(qc, aer_simulator, noise_model=noise_model,basis_gates=basis_gates)
```
does it not fail.

**Comment by 1ucian0** [comment]
I'm not able to reproduce this issue @bjader Can we come up with a minimal failing example?

This seems like working:
```python
from qiskit import QuantumRegister, QuantumCircuit, ClassicalRegister, execute
from qiskit import IBMQ
IBMQ.load_accounts()

tokyo = IBMQ.get_backend('ibmq_20_tokyo')

properties = tokyo.properties()
coupling_map = tokyo.configuration().coupling_map

qr = QuantumRegister(5)
cr = ClassicalRegister(1)
qc = QuantumCircuit(qr, cr)

qc.h(qr[0])
qc.x(qr[0])
qc.cx(qr[0], qr[1])
qc.x(qr[0])
qc.cx(qr[0], qr[1])
qc.h(qr[0])

execute(qc, tokyo, coupling_map=coupling_map)
```

**Comment by ajavadia** [comment]
The issue says that when executing on Tokyo the error does not arise. Try executing on the simulator, with the Tokyo coupling map.

**Comment by bjader** [comment]
Hi @1ucian0 I can concur that your example does not fail for me either - but it should be noted in your example you are running on the real tokyo device. In my original post the failure I'm reporting occurs when using the Tokyo coupling map on the qasm simulator.

Here is a more minimal example of my original code with only the qubit initialization left, that does fail for me:
```python
from qiskit import QuantumRegister, QuantumCircuit, ClassicalRegister, execute
from qiskit import IBMQ
IBMQ.load_accounts()

tokyo = IBMQ.get_backend('ibmq_20_tokyo')
aer_simulator = Aer.get_backend('qasm_simulator')

properties = tokyo.properties()
coupling_map = tokyo.configuration().coupling_map

qr = QuantumRegister(5)
cr = ClassicalRegister(1)
qc = QuantumCircuit(qr, cr)

ground_state = [0.00000000e+00 + 0.j, 0.00000000e+00 + 0.j, 0.00000000e+00 + 0.j,
               0.00000000e+00 + 0.j, 0.00000000e+00 + 0.j, 3.71748034e-01 + 0.j,
                        -6.01500955e-01 + 0.j, 0.00000000e+00 + 0.j, 0.00000000e+00 + 0.j,
                        -6.01500955e-01 + 0.j, 3.71748034e-01 + 0.j, 1.33322273e-17 + 0.j,
                        0.00000000e+00 + 0.j, 0.00000000e+00 + 0.j, 0.00000000e+00 + 0.j,
                        0.00000000e+00 + 0.j]

qc.initialize(ground_state, [qr[1], qr[2], qr[3], qr[4]])

execute(qc, aer_simulator, coupling_map=coupling_map)
```

**Comment by ajavadia** [comment]
@bjader which version of qiskit terra are you running? Are you running from master or pip 0.7?

**Comment by bjader** [comment]
@ajavadia I'm using a conda virtual environment based on the environment.yaml defined in the qiskit tutorials. Details from `conda list`:

```
Name                    Version                     Build   Channel
qiskit                    0.8.0                    pypi_0    pypi
qiskit-aer                0.1.1                    pypi_0    pypi
qiskit-aqua               0.4.1                    pypi_0    pypi
qiskit-chemistry          0.4.2                    pypi_0    pypi
qiskit-ignis              0.1.0                    pypi_0    pypi
qiskit-terra              0.7.1                    pypi_0    pypi
```

**Comment by ajavadia** [comment]
Ok thanks. On master, I get failures running both on simulator and device. I am investigating.

**Comment by ajavadia** [comment]
Ok, this issue is actually no longer present on master now. 

The problem is finding a good initial layout (if this is done, the chance that the swapper succeeds is increased). The issue was that in version 0.7.1, the code to find the layout was only being called for the device, not the simulator. This is now fixed on master.

However, there is a tangential issue which still makes the original code not runnable on master. The `initialize()` instruction has temporarily become less optimized, which must be fixed (#1943). Until this is done, using the `initialize()` instruction still fails.

However, to demonstrate that the original **circuit** is runnable, you can manually code it like this:
```
q = QuantumRegister(5, 'q')
circ = QuantumCircuit(q)

circ.ry(1.6, q[4])
circ.cx(q[4], q[3])
circ.ry(1.6, q[3])
circ.cx(q[4], q[3])
circ.ry(1.6, q[3])
circ.cx(q[4], q[3])
circ.rz(0.79, q[3])
circ.cx(q[4], q[3])

circ.cx(q[4], q[2])
circ.ry(0.23, q[2])
circ.cx(q[3], q[2])
circ.ry(-0.79, q[2])
circ.cx(q[4], q[2])
circ.ry(-0.23, q[2])
circ.cx(q[3], q[2])
circ.ry(0.79, q[2])
circ.cx(q[4], q[2])
circ.rz(0.79, q[2])
circ.cx(q[3], q[2])
circ.cx(q[4], q[2])
circ.rz(-0.79, q[2])
circ.cx(q[3], q[2])

circ.cx(q[4], q[1])
circ.cx(q[2], q[1])
circ.cx(q[3], q[1])
circ.ry(-0.79, q[1])
circ.cx(q[2], q[1])
circ.ry(-0.79, q[1])
circ.cx(q[4], q[1])
circ.cx(q[2], q[1])
circ.cx(q[3], q[1])
circ.ry(0.79, q[1])
circ.cx(q[2], q[1])
circ.rz(0.79, q[1])
circ.cx(q[4], q[1])
circ.rz(-0.79, q[1])
circ.cx(q[2], q[1])
circ.cx(q[3], q[1])
circ.rz(-0.79, q[1])
circ.cx(q[2], q[1])
circ.cx(q[4], q[1])
circ.rz(0.79, q[1])
circ.cx(q[2], q[1])
circ.cx(q[3], q[1])
circ.rz(0.79, q[1])
circ.cx(q[2], q[1])

new_circ = transpile(circ, aer_simulator, coupling_map=coupling_map, seed_mapper=10)
```

before mapping:
![image](https://user-images.githubusercontent.com/8622381/55635131-f988ad00-578d-11e9-951a-73add2c1289d.png)

after mapping:
![image](https://user-images.githubusercontent.com/8622381/55635140-fdb4ca80-578d-11e9-9b40-db620a852e3a.png)


I'll keep this issue open until after #1943 is fixed, to make sure everything works.

**Comment by bjader** [comment]
Hi @ajavadia, thanks for your hard work and for finding the root of this problem. Looking forward to the fix in the next release.
